app/models/snp.rb

```diff
@@ -19,6 +17,34 @@ class Snp < ActiveRecord::Base
 
   after_create :default_frequencies
 
+  default_scope { select(column_names - ['genotypes']) }
+
+  def genotypes
+    genotype_ids = self.class.unscoped
+                             .select('unnest(akeys(genotypes)::int[])')
+                             .where(id: id)
+    Genotype.where(id: genotype_ids)
+  end
+
+  def genotypes_count
+    @genotype_count ||= self.class.where(id: id)
+                                  .pluck('array_length(akeys(genotypes), 1)')
```

Comment:
Use 2 (not 30) spaces for indenting an expression in an assignment spanning multiple lines.